APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.h

```diff
@@ -89,7 +89,18 @@
  
  @param     filename                Filename for the json data to be included without path extension
  */
-- (void)insertIntoArchive:(NSDictionary *)dictionary filename: (NSString *)filename;
+- (void)insertDictionaryIntoArchive:(NSDictionary *)dictionary filename: (NSString *)filename;
```

Comment:
Yeah, I got very confused about why it wasn't crashing on attempting to push a dictionary when expecting NSData.